src/router/router.tsx

```diff
@@ -0,0 +1,38 @@
+import { createBrowserRouter } from 'react-router';
+
+import { FallbackUi } from '@/components/FallbackUi/FallbackUi';
+import { RedirectWithSearch } from '@/components/InitialRedirect/InitialRedirect';
+import { Root } from '@/pages/Root';
+import { AboutPage, AnimeDetails, HomePage, NotFoundPage } from '@/router/lazyElements';
+import { animeDetailsLoader, animeListLoader } from '@/router/loaders';
+
+export const routes = [
+  {
+    path: '/',
+    element: <Root />,
+    errorElement: <FallbackUi />,
+    children: [
+      {
+        index: true,
+        element: <RedirectWithSearch />,
+      },
+      {
+        path: ':page',
+        element: <HomePage />,
+        loader: animeListLoader,
+        errorElement: <FallbackUi buttonMessage="Back To List" />,
```

Comment:
fix warning "No `HydrateFallback` element provided to render during initial hydration"
```suggestion
        errorElement: <FallbackUi buttonMessage="Back To List" />,
        HydrateFallback: () => <div role="status">Loading...</div>,
```
minimal fix adds a simple loading div with role=\"status\" which matches the loading state pattern already used in your tests.